app/src/main/java/com/example/cinemapp/ui/main/collections/CollectionsFragment.kt

```diff
@@ -51,15 +47,11 @@ class ProfileFragment : Fragment() {
         super.onStart()
         viewModel.getInitialData()
         setupOnClick()
-        observeSignOutEvent()
         setupTabs()
 
         observeFlowSafely(viewModel.state) {
-            it.accountDetails?.let { accountDetails ->
-                setupDetails(accountDetails)
-            }
             adapter.setMovies(it.movies)
-            setupLoadingVisibility(it.isLoading, it.accountDetails?.let { true } ?: false)
+            setupLoadingVisibility(it.isLoading, (activity as MainActivity).isSignedIn())
```

Comment:
while this cast would always succeed i would advise you to use safe cast `as?` that returns null if the cast fails instead of a crash, depending on the usecase